doc/whats-new.rst

```diff
@@ -38,6 +38,10 @@ Bug fixes
 Documentation
 ~~~~~~~~~~~~~
 
+- Updated the release guide for developers. Now accounts for actions that are automated via github
+  actions. Pull (:pull:`5274`).
+  By `Tom Nicholas <https://github.com/TomNicholas>`_.
```

Comment:
since that's not included in the docs, should we move this to internals? Otherwise we can move `HOW_TO_RELEASE.md` to a "contributing guide for core devs" and also include the guidelines for pushing to `master` / `stable`.

Reply:
Moved to internals.